sale_coupon_multiplier_free_product/models/sale_coupon.py

```diff
@@ -4,10 +4,13 @@
 
 
 class SaleCoupon(models.Model):
-    _inherit = "sale.coupon"
+    _inherit = "coupon.coupon"
 
-    def _check_coupon_code(self, order):
-        message = super()._check_coupon_code(order)
+    def _check_coupon_code(self, order_date, partner_id, **kwargs):
+        order = kwargs.get("order")
+        message = super()._check_coupon_code(
+            order.date_order.date(), order.partner_id.id, order=order
```

Comment:
This isn't right. You should pass the args as they come not assuming the values in your override